workitem/workitem_repository.go

```diff
@@ -701,50 +702,201 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
 	return &result, nil
 }
 
-// GetCountsPerIteration fetches WI count from DB and returns a map of iterationID->WICountsPerIteration
-// This function executes following query to fetch 'closed' and 'total' counts of the WI for each iteration in given spaceID
-// 	SELECT iterations.id as IterationId, count(*) as Total,
-// 		count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed
-// 		FROM "work_items" left join iterations
-// 		on fields@> concat('{"system.iteration": "', iterations.id, '"}')::jsonb
-// 		WHERE (iterations.space_id = '33406de1-25f1-4969-bcec-88f29d0a7de3'
-// 		and work_items.deleted_at IS NULL) GROUP BY IterationId
-func (r *GormWorkItemRepository) GetCountsPerIteration(ctx context.Context, spaceID uuid.UUID) (map[string]WICountsPerIteration, error) {
-	defer goa.MeasureSince([]string{"goa", "db", "workitem", "getCountPerIteration"}, time.Now())
+func (r *GormWorkItemRepository) getAllIterationWithCounts(ctx context.Context, db *gorm.DB, spaceID uuid.UUID) (map[string]WICountsPerIteration, error) {
+	var allIterations []uuid.UUID
+	db.Pluck("id", &allIterations)
+	iterationTable := iteration.Iteration{}
+	iterationTableName := iterationTable.TableName()
+	iterationWithWICount := fmt.Sprintf(`
+	SELECT count(*) AS Total,
+		count(CASE fields->>'system.state'
+					WHEN 'closed' THEN '1'
+					ELSE NULL
+				END) AS Closed,
+		fields->>'system.iteration' AS iterationID
+	FROM %s wi
+	WHERE fields->>'system.iteration' IN
```

Comment:
I am struggling with using Index here because it is `IN` clause
I am following https://www.postgresql.org/docs/9.6/static/datatype-json.html
Tried creating `expression index` like - 
` CREATE INDEX trial_index ON work_items USING gin((fields -> 'system.state'));`
And then wanted to query like -
`select * from work_items where fields->'system.state' @> '{"system.state": ["open", "closed"]}'`

Above queries are for example, in reality, I will be using `system.iteration` field and not `system.state`